src/audio/module_adapter/module/volume/volume.c

```diff
@@ -361,6 +362,7 @@ static void reset_state(struct vol_data *cd)
 	cd->vol_ramp_frames = 0;
 	cd->vol_ramp_elapsed_frames = 0;
 	cd->sample_rate = 0;
+	cd->copy_needed = true;
```

Comment:
Is it ensured that volume_ramp() is always executed before first vol_sX() function? Would it be good to set this also in volume_prepare()?

Reply:
Actually to new() or IPC4 equivalent if the state set in reset should be same as new().